src/ansi-c/c_typecheck_code.cpp

```diff
@@ -911,7 +911,15 @@ void c_typecheck_baset::typecheck_return(codet &code)
     {
       // gcc doesn't actually complain, it just warns!
       if(follow(code.op0().type()).id()!=ID_empty)
+      {
+        warning().source_location = code.source_location();
+
+        warning() << "Function has return void but a return statement returning ";
```

Comment:
We seem to be starting all warnings with a lowercase character (i.e., use "function")

Also, that line is rather long, consider breaking it earlier.